Assets/Scripts/CoinCounter.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using UnityEngine;
+
+namespace DefaultNamespace
+{
+    public class CoinCounter : MonoBehaviour
+    {
+        public Action<int> CoinsUpdated;
+
+        private void Start()
+        {
+            _coins = PlayerPrefs.GetInt("coins", 0);
+        }
+
+        private void OnDisable()
+        {
+            PlayerPrefs.SetInt("coins", _coins);
```

Comment:
you should not rely on OnDisable to handle something when the game ends. It will not work on Android, for example (because Android apps get killed rather than gracefully quit)